akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -749,6 +749,46 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
         (s: S, out: Out) => f.apply(s, out).toScala,
         (s: S) => onComplete.apply(s).asScala))
 
+  /**
+   * Transform each stream element with the help of a resource.
+   *
+   * The resource creation function is invoked once when the stream is materialized and the returned resource is passed to
+   * the mapping function for mapping the first element. The mapping function returns a mapped element to emit
+   * downstream. The returned `T` MUST NOT be `null` as it is illegal as stream element - according to the Reactive Streams specification.
+   *
+   * The `close` function is called only once when the upstream or downstream finished, You can do some clean-up here,
```

Comment:
```suggestion
   * The `close` function is called only once when the upstream or downstream finishes or fails, You can do some clean-up here,
```

And the same for the other Scaladoc sections